src/Apps/Order/Routes/__tests__/Payment.test.tsx

```diff
@@ -282,25 +278,45 @@ describe("Payment", () => {
       },
     }
 
-    stripeMock.createToken.mockReturnValue({ then: func => func(stripeError) })
+    stripeMock.createToken.mockReturnValue(Promise.resolve(stripeError))
 
     const paymentRoute = getWrapper(testProps)
     fillAddressForm(paymentRoute, validAddress)
 
+    expect(paymentRoute.text()).not.toContain("Your card number is invalid.")
+
+    paymentRoute.find(ContinueButton).simulate("click")
+
+    // flush the promise queue
+    await Promise.resolve()
+
+    expect(paymentRoute.text()).toContain("Your card number is invalid.")
+  })
+
+  it("shows an error modal when CreateToken raises an error", async () => {
+    stripeMock.createToken.mockReturnValue(
+      Promise.reject(new Error("something failed"))
+    )
+    const paymentRoute = getWrapper(testProps)
+    fillAddressForm(paymentRoute, validAddress)
     paymentRoute.find(ContinueButton).simulate("click")
 
+    // flush promise queue
+    await Promise.resolve()
+    await Promise.resolve()
```

Comment:
Why do we have this twice?

Reply:
jest has an internal promise queue. when you do 

    await Promise.resolve()

it pushes that new promise to the end of the queue and execution continues when the queue gets emptied up to and including that new promise.

However, if during that time more promises get added to the queue, then those will still be unresolved. If your test depends on those extra promises being resolved and some effects happening, you will need to flush the queue again. And maybe again. And so on. Depends how long your promise chains get. 

Something like this https://github.com/facebook/jest/issues/2157#issuecomment-279171856 might mitigate that. Worth trying!